### DISCUSSION TITLE
Duplicated ranking values in gsea?

### DISCUSSION THREAD

**Discussion by BelanaZ** [zqfang/GSEApy#217] [Q&A]
If duplicated values are present in the gene ranking for preranked GSEA, this message is alerted:
"Duplicated values found in preranked stats: X% of genes. The order of those genes will be arbitrary, which may produce unexpected results"
Is there any information available on what happens for the standard GSEA function if duplicate values are in the ranking? Is the order arbitrary as well or is it always handled the same, e.g. by first sorting by value and then by gene name?

**Comment by zqfang** [comment]
Sorry, I don't have information about the GSEA(broad) on duplicated ranking values.

**Reply by BelanaZ** [reply]
Sorry, for the confusion. I was actually referring to the GSEA class you have implemented here, not the version developed by the Broad Institute. How are duplicate ranking values handled? In case you were also referring to GSEApy's GSEA algorithm, then kindly disregard this. @zqfang

**Reply by zqfang** [reply]
It is the same order as the input ranking list (for preranked list).

you could check the ordering by the `ranking` attr 
```
prk = gseapy.prerank(...)
# .ranking
prk.ranking
```
